base/array.jl

```diff
@@ -863,22 +863,9 @@ function indexin(a::AbstractArray, b::AbstractArray)
     [get(bdict, i, 0) for i in a]
 end
 
-# findin (the index of intersection)
-function findin(a, b::UnitRange)
-    ind = Array(Int, 0)
-    f = first(b)
-    l = last(b)
-    for i = 1:length(a)
-        if f <= a[i] <= l
-            push!(ind, i)
-        end
-    end
-    ind
-end
-
 function findin(a, b)
     ind = Array(Int, 0)
-    bset = Set(b)
+    bset = unique(b)
     @inbounds for i = 1:length(a)
         a[i] in bset && push!(ind, i)
```

Comment:
bset is probably not a good variable name if it can be an array.

This `in` will be O(length(bset)) rather than O(1) right? I agree that's fine but only IF you can explicitly pass `findin(a, Set(b))` as an optimization for large arrays b. (Assuming `unique ∘ Set` returned a Set as discussed in your other issue.) IMO This should be noted in the docstring... once it's true.